docs/getting-started-guides/docker-multinode.md

```diff
@@ -74,54 +78,165 @@ This pattern is necessary because the `flannel` daemon is responsible for settin
-`Worker done!`
+And on Master node, you can see extra master containers running:
+
+```console
+k8s_scheduler.xxx
+k8s_apiserver.xxx
+k8s_controller-manager.xxx
+```
 
-See [here](docker-multinode/worker.md) for detailed instructions explanation.
+As we use `hyperkube` image to run k8s, we **do not** need to compile binaries, please download and extract `kubectl` binary from [releases page](https://github.com/kubernetes/kubernetes/releases).
 
-## Deploy a DNS
+At last, use `$ kubectl get nodes` to see if all of your nodes are ready.
+
+```console
+$ kubectl get nodes
+NAME            LABELS                                 STATUS
+10.10.102.150   kubernetes.io/hostname=10.10.103.150   Ready
+10.10.102.153   kubernetes.io/hostname=10.10.102.153   Ready
+```
+
+### Deploy a DNS
 
 See [here](docker-multinode/deployDNS.md) for instructions.
 
-## Testing your cluster
+Then you can run Kubernetes [guest-example](../../examples/guestbook/) to build a Redis backend cluster on the k8s．
+
+## Customize
+
+One of the biggest benefits of using Docker to run Kubernetes is users can customize the cluster freely before you running `kube-up.sh`:
+
+### master
+
+You need to enable **customized master mode** before deploy:
+
+```sh
+export MASTER_CONF=yes
+```
+
+The configuration file of Master locates in `cluster/images/hyperkube/master-multi.json`, which will be mounted as volume for Master Pod to consume, you can modify it freely **before deploying**.
+
+In this mode, you can even change the configuration of Master after the deployment has done, see:
+
+1. Login the Master node
+2. Change the content in `~/docker/kube-config/master-multi.json`
+
+kubelet will auto-restart the affected master pod.
+
+### kubelet
+
+Except a few basic options defined in `kube-deploy/master.sh|node.sh`, you can customize the `docker/kube-config/kubelet.env` freely to add or update `kubelet` options **before deploying**.
+
+## Tear Down
+
+In `cluster/` directory:
+
+```sh
+export NODES="vcap@10.10.102.150 vcap@10.10.102.152"
+export KUBERNETES_PROVIDER=docker ./kube-down.sh
+```
+
+## Trouble shooting
+
+Although using docker to deploy k8s is much simpler than ordinary way, here're some tips to follow if there's any trouble.
+
+### What did the scripts do?
+
+1. Start a bootstrap daemon
+2. Start `flannel` on every node's bootstrap daemon, `etcd` on Master's bootstrap daemon
+3. Start `kubelet` & `proxy` containers by using `hyperkube` image on every node
+4. `kubelet` on the Master node will start master Pod (contains `api-server`, `controller-manager` & `scheduler`)from a built in `json` file in hyperkube image.
+
+See [docker-multinode/master.md](docker-multinode/master.md) and [docker-multinode/worker.md](docker-multinode/worker.md) for detailed instructions explanation.
+
+### Useful tips
+
+1. Make sure you have access to the images stored in `gcr.io`, otherwise, you need to manually load `hyperkube` image into your docker daemon and `etcd` into docker bootstrap daemon.
```

Comment:
I meant documentation of these (at least):
 - `gcr.io/google_containers/hyperkube:v$K8S_VERSION`
 - `gcr.io/google_containers/etcd:$ETCD_VERSION`
 - `quay.io/coreos/flannel:$FLANNEL_VERSION`

So even if one want to use another image, one have to `docker tag` them to that name.